### PR TITLE
[Dashboard] Migrate Logs page to use state api.

### PR DESCRIPTION
**Pull request by alanwguo**
## Why are these changes needed?
1. Migrates to use state-api, to unify behavior with CLI and UI
2. Delete log_proxy API, it's legacy and has some security issues.

## Checks

- [x] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: dashboard/modules/log/log_agent.py
Comment (rkooo567) [inline dashboard/modules/log/log_agent.py:299]: does it work with nested directory inside log dir btw?

Comment (rkooo567) [reply dashboard/modules/log/log_agent.py:299]: (it doesn't have to work imo, but i'd like to just know)

File: dashboard/client/src/pages/log/Logs.tsx
Comment (rkooo567) [inline dashboard/client/src/pages/log/Logs.tsx:79]: Btw, this doesn't seem to work when I used it

**Review by rkooo567 — APPROVED** [review]
It seems good to me in my first look! I think for the frontend code, it'd be great other experience team members take a look at it. (I will wait for their approval)

File: dashboard/modules/log/log_agent.py
Comment (alanwguo) [reply dashboard/modules/log/log_agent.py:299]: yup, I tested this

File: dashboard/client/src/pages/log/Logs.tsx
Comment (alanwguo) [reply dashboard/client/src/pages/log/Logs.tsx:79]: oh shoot, I didn't test this. let me test thi

Comment (alanwguo) [reply dashboard/client/src/pages/log/Logs.tsx:79]: Fixed!

File: dashboard/client/src/pages/log/Logs.tsx
Comment (brycehuang30) [inline dashboard/client/src/pages/log/Logs.tsx:188]: To confirm, the files listed by `listStateApiLogs` could be either filepath or filename, so that's why we need this branching, right?

**Review by brycehuang30 — APPROVED** [review]
Frontend code LGTM! Though I have limited context and I didn't test-run the code.

**Comment by pcmoritz** [comment]
@alanwguo Thanks a lot for doing this -- there is still a failing test, can you fix it? https://buildkite.com/ray-project/premerge/builds/13040#018c197d-5f5b-4abd-999b-08be7f0af010

**Review by pcmoritz — APPROVED** [review]
I tried it locally and it is working :)

**Comment by alanwguo** [comment]
verified that multi-node log viewing continues to work. will fix the faiiling test

**Review by pcmoritz — APPROVED** [review]

**Comment by alanwguo** [comment]
verified again with latest changes. 
This should be good to merge as soon as tests pass

**Pull request merged**
